src/Auth/HttpBasicAuth.php

```diff
@@ -0,0 +1,111 @@
+<?php
+namespace Yiisoft\Yii\Web\Auth;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Yiisoft\Yii\Web\User\IdentityInterface;
+use Yiisoft\Yii\Web\User\IdentityRepositoryInterface;
+
+/**
+ * HttpBasicAuth is an action filter that supports the HTTP Basic authentication method.
+ *
+ * > Tip: In case authentication does not work like expected, make sure your web server passes
+ * username and password to `$_SERVER['PHP_AUTH_USER']` and `$_SERVER['PHP_AUTH_PW']` variables.
```

Comment:
We can not rely on `$_SERVER` anymore and should get all the info from PSR-7 ServerRequestInterface headers.